workspaces/src/cargo/mod.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::anyhow;
+use cargo_metadata::Message;
+use std::fmt::Debug;
+use std::path::Path;
+use std::process::{Command, Stdio};
+use tracing::debug;
+
+pub fn build_cargo_project<P: AsRef<Path> + Debug>(
+    project_path: P,
+) -> anyhow::Result<Vec<Message>> {
```

Comment:
prob not ideal to expose `cargo_metadata` types in our API

Reply:
Yeah, I should have made this `pub(crate)` but since I moved all compilation logic into `cargo`, this can be private now